src/pg_cron.c

```diff
@@ -332,6 +332,13 @@ pg_cron_sigterm(SIGNAL_ARGS)
 	{
 		SetLatch(&MyProc->procLatch);
 	}
+
+	if (!proc_exit_inprogress)
+	{
+		InterruptPending = true;
+		ProcDiePending = true;
+	}
+	CHECK_FOR_INTERRUPTS();
```

Comment:
this CHECK_FOR_INTERRUPTS probably does not belong here, since it would cause a longjmp from the signal handler, which can result in undefined behaviour afaik